### ISSUE TITLE
helm: Allow for iRSA

### ISSUE THREAD

**Issue by jwitko** [clastix/kamaji-etcd#52]
The helm chart currently only allows for hard-coded environment variable based AWS IAM access key and secret key input.  It does not allow for not-specifying these and letting iRSA handle it via the kubernetes service account.

**Comment by prometherion** [comment]
I'm not an AWS expert here, please, correct me if I'm wrong, and if I'm missing the context that should be about the following environment variables:

https://github.com/clastix/kamaji-etcd/blob/0536a5452d074022edfd5647040d14ee30539aaf/charts/kamaji-etcd/values.yaml#L100-L103

https://github.com/clastix/kamaji-etcd/blob/0536a5452d074022edfd5647040d14ee30539aaf/charts/kamaji-etcd/values.yaml#L105-L110

If I understood correctly, we should make these ones optional to let the S3 client interact using the Kubernetes Service Account.

**Comment by jwitko** [comment]
Yes and also changing the conditionals around these to not be supplied:
https://github.com/clastix/kamaji-etcd/blob/0536a5452d074022edfd5647040d14ee30539aaf/charts/kamaji-etcd/templates/etcd_cronjob_backup.yaml#L76-L89

**Comment by bsctl** [comment]
@kvaps for commenting

**Comment by kvaps** [comment]
I don't mind, but we already switched to using
https://github.com/aenix-io/etcd-operator

Currently it already implements the same functionality as this Helm chart, except periodic defragmentation and backups which can simple done with cronjobs.

We're working on submitting this project to sig-etcd.
The working group already established:
- https://github.com/kubernetes/community/pull/7917

This week we had a first meeting with SIGs, where we researched features needed by the community to prepare the roadmap.
